### PR TITLE
BFF sample with front and back end hosts

### PR DESCRIPTION
**Pull request by josephdecock**

### CODE REVIEW

File: IdentityServer/v6/BFF/SplitHosts/FrontendHost/wwwroot/session.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      console.log("user claims", claims);
+
+      let logoutUrlClaim = claims.find(
+        (claim) => claim.type === "bff:logout_url"
+      );
+      if (logoutUrlClaim) {
+        document.getElementById("logoutLink").href =
+          "https://localhost:5010" +
+          logoutUrlClaim.value +
+          "&returnUrl=https://localhost:5011/";
```
Comment (josephdecock) [inline IdentityServer/v6/BFF/SplitHosts/FrontendHost/wwwroot/session.js:28]: This interpolation only works if there are query parameters in the logoutUrlClaim. I think that is always the case - but is it possible that there wouldn't be a sid and thus no query parameter? I could go read the source and find out if that can ever happen, but either way, this code is coupled really tightly to the specifics of the logoutUrlClaim and I don't love that.

File: IdentityServer/v6/BFF/SplitHosts/BackendHost/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddCors(opt => 
+            {
+                opt.AddDefaultPolicy(policy =>
+                {
+                    policy.WithOrigins("https://localhost:5011")
+                        .AllowAnyHeader()
+                        .AllowAnyMethod()
```
Comment (josephdecock) [inline IdentityServer/v6/BFF/SplitHosts/BackendHost/Startup.cs:21]: This is very permissive CORS policy. Would a more locked down set of policies be helpful or just distracting?

**Comment by brockallen** [comment]
We'll wait on merging this until we flesh out the discussion: https://github.com/DuendeSoftware/docs.duendesoftware.com/issues/227

File: IdentityServer/v6/BFF/SplitHosts/BackendHost/Startup.cs (hunk 2 above)
Comment (leastprivilege) [reply IdentityServer/v6/BFF/SplitHosts/BackendHost/Startup.cs:21]: yes - it needs to explicitly allow only the required headers + the X-CSRF header.

Comment (josephdecock) [reply IdentityServer/v6/BFF/SplitHosts/BackendHost/Startup.cs]: Done!

**Review by brockallen — APPROVED** [review]

**Pull request merged**